solc_select/solc_select.py

```diff
@@ -205,6 +320,15 @@ def get_url(version: str = "", artifact: str = "") -> (str, str):
 def switch_global_version(version: str, always_install: bool, silent: bool = False) -> None:
     if version == "latest":
         version = get_latest_release()
+
+    # Check version against platform minimum even if installed
+    if version != "latest" and Version(version) < Version(
+        EARLIEST_RELEASE[soliditylang_platform()]
+    ):
+        raise argparse.ArgumentTypeError(
+            f"Invalid version - only solc versions above '{EARLIEST_RELEASE[soliditylang_platform()]}' are available"
+        )
+
```

Comment:
This is already validated through `valid_version` in the arg parser, what motivated this additional check here?

```python
    parser_use.add_argument(
        USE_VERSION, help="solc version you want to use (eg: 0.4.25)", type=valid_version, nargs="?"
    )
```